Issues/310, Issues/195 (#320)
V4Fire/Client: src/core/dom/in-view/README.md

# core/dom/in-view

> This module provides an API to track elements entering or leaving the viewport.

Reviewer: ```suggestion
This module provides API to track elements entering or leaving the viewport.
```

| ---------- |----------------------------------------------------------------------------------| -------------------- |------------- |
| `callback` | Invoked when an element stands in the viewport more than the specified delay     | `Observable` element | `Observable` |
| `onEnter`  | Invoked when an element passed the specified `threshold` and enters the viewport | `Observable` element | `Observable` |
| `onLeave`  | Invoked when an element passed the specified `threshold` and leaves the viewport | `Observable` element | `Observable` |

## Usage

```ts
import { InView } from 'core/dom/in-view';

export default class bFullScreenView extends iBlock implements iLockPageScroll {
  @hook('mounted')
  initInView(): void {
    InView.observe(this.$el, {
      threshold: 0.7,
      delay: 2000,
      once: true,
      callback: () => this.emit('elementInViewport'),
      onEnter: () => this.emit('elementEnterViewport'),
      onLeave: () => this.emit('elementLeaveViewport')
    })
  }
}
```

Also, the module can take an array of options for multiple observing.

> **Notice:** If you wanna observe a single element with multiple observers then the observers should have different thresholds.

Reviewer: ```suggestion
**Notice:** If you want to observe a single element with multiple observers, the observers must have different thresholds.
```

# core/dom/in-view

> This module provides a directive to track elements entering or leaving the viewport.

Reviewer: Это ведь не так. Этот модуль предоставляет API. Я уже про это выше писал. Поправь описание там и тут.
